### PR TITLE
Fix duplicate tests to assert on stdout logs

### PR DESCRIPTION
**Pull request by codyborders**
for [Dev Agent Session](https://app.datadoghq.com/code/60870c3e-9ab7-4af7-b63c-26319fcefa5f) identified by [Trace Investigation](https://support-admin.us1.prod.dog/admin/switch_handle_get/org_id/240457?next_url=%2Fapm%2Ftrace%2F6934c39700000000613a2f56ca86a332%3FspanID%3D18374819542212200335%26trace-assistant-investigation-id%3D726fa3c3-28d2-42f0-9619-4b563472517e%26trace-assistant-panel-open%3Dtrue&reason=incident-31285)

---

Summary
- Update duplicate-related tests to validate JSON logs emitted to stdout instead of using caplog, since the project logger writes to stdout and bypasses caplog handlers.

Changes
- tests/test_scrape_github_trending.py:
  - test_scrape_github_trending_skips_duplicates_precheck: replace caplog with capsys and assert "scraper.skip_duplicate" in captured stdout.
  - test_scrape_github_trending_integration_duplicate_with_temp_db: replace caplog with capsys and assert "scraper.skip_duplicate" in captured stdout.

Impact
- Stabilizes tests regardless of logger handler configuration; functional behavior unchanged.

Testing
- Format/lint pass; duplicate tests now assert against stdout as requested.

### CODE REVIEW

File: tests/test_scrape_github_trending.py [hunk 1]
```diff
@@ -102,3 +102,88 @@ def test_fake_response_raises_on_error():
     response = FakeResponse(status_code=500)
     with pytest.raises(Exception):
         response.raise_for_status()
+
+
+def test_scrape_github_trending_skips_duplicates_precheck(monkeypatch):
```
Comment (codyborders) [inline tests/test_scrape_github_trending.py]: Can we add type hints to the new tests/helpers to keep tests aligned with PYTHON.md? Right now the new test functions and stubs are unannotated.

File: tests/test_scrape_github_trending.py [hunk 2]
```diff
@@ -102,3 +102,88 @@ def test_fake_response_raises_on_error():
...
+
+    # Classifier marks it as a devtool
+    def fake_classify(candidates):
+        candidates = list(candidates)
+        return {item["id"]: True for item in candidates}
+
+    monkeypatch.setattr("scrape_github_trending.classify_candidates", fake_classify)
+    monkeypatch.setattr("scrape_github_trending.get_devtools_category", lambda *args, **kwargs: None)
+
+    # is_duplicate pre-check forces skip
```
Comment (codyborders) [inline tests/test_scrape_github_trending.py]: Since we stub `is_duplicate` here, we never exercise the real SQLite duplicate guard. Could we add an integration-style test with a temp DB to verify duplicates are skipped without calling `save_startup`, so DB-level regressions are caught?

**Comment by codyborders** [comment]
I reran `pytest tests/test_scrape_github_trending.py` on this branch (`842260a`) and two duplicate-related tests failed. The skip logic works, but the DEBUG log `scraper.skip_duplicate` never reaches `caplog` because the logger is configured at INFO with propagate=False. `caplog.set_level("DEBUG")` alone doesn’t lower the logger; set the logger level (e.g., `caplog.set_level("DEBUG", logger="devtools.scraper.github_trending")`) or assert on an INFO-level signal instead. Once the logger is lowered for the test, the assertions should pass.
